wkcuber/compress.py

```diff
@@ -80,13 +81,21 @@ def compress_mag(source_path, layer_name, target_path, mag: Mag, jobs):
     target_mag_path = path.join(target_path, layer_name, str(mag))
     logging.info("Compressing mag {0} in '{1}'".format(str(mag), target_mag_path))
 
-    with open_wkw(source_wkw_info) as source_wkw, ParallelExecutor(jobs) as pool:
+    with open_wkw(source_wkw_info) as source_wkw:
         source_wkw.compress(target_mag_path)
-        for file in source_wkw.list_files():
-            rel_file = path.relpath(file, source_wkw.root)
-            pool.submit(compress_file_job, file, path.join(target_mag_path, rel_file))
-
-    logging.info("Mag {0} succesfully compressed".format(str(mag)))
+        with get_executor_for_args(args) as executor:
+            futures = []
+            for file in source_wkw.list_files():
+                rel_file = path.relpath(file, source_wkw.root)
+                futures.append(
+                    executor.submit(
+                        compress_file_job, file, path.join(target_mag_path, rel_file)
+                    )
+                )
+
+            wait_and_ensure_success(futures)
```

Comment:
Would a `executor.map(...)` catch potential errors?

Reply:
Yes, it would. However, the map interface seemed a bit cumbersome to me since the "preparation code" for one submit is > 10 lines long. We could make a for-loop which pushes the task descriptors into a list and then map over that. However, I think it makes sense to establish a good precedent case on how to properly use `submit` for the concurrent.futures.executor interface. Also, it makes the success-checking explicit. For `map`, the success checking is a quite unobvious side-effect, in my opinion.

Reply:
Fine with me.